vendor/github.com/vmware/go-vcloud-director/v2/govcd/vm.go

```diff
@@ -850,3 +851,144 @@ func (vm *VM) SetGuestCustomizationSection(guestCustomizationSection *types.Gues
 
 	return vm.GetGuestCustomizationSection()
 }
+
+func (vm *VM) AddDisk(diskData *types.DiskSettings) (string, error) {
```

Comment:
To avoid possible confusion with current independent disk, I suggest naming all functions more explicitly with word "Internal" in it, like `AddInternalDisk(...)`